src/goto-symex/goto_symex_state.cpp

```diff
@@ -797,3 +797,34 @@ void goto_statet::output_propagation_map(std::ostream &out)
     out << name_value.first << " <- " << format(name_value.second) << "\n";
   }
 }
+
+void goto_symex_statet::add_object(
+  ssa_exprt &ssa,
+  std::size_t l1_index,
+  const namespacet &ns)
+{
+  framet &frame = top();
+
+  const irep_idt l0_name = ssa.get_identifier();
+
+  // save old L1 name, if any
+  auto c_it = level1.current_names.find(l0_name);
```

Comment:
Do a single `emplace` and then adjust if it already exists, rather than the find / emplace double-lookup pattern?

Reply:
:+1:

Reply:
Done.